### DISCUSSION TITLE
Is there a way to generate custom errors when doing custom serialization?

### DISCUSSION THREAD

**Discussion by scalpel4k** [stephenberry/glaze#1145] [General]
Hi Steven,

I'm having some JSON data that I have to validate during (de)serialization using `to_json<>/from_json<>` specializations.

I'd like to use the error object in a way that it tells me a bit more about parsing problems. `current_file` seems like the only field that I can somehow shoehorn into giving me some extra info, and the current error_codes do not actually match any of my problems (e.g. cardinality errors).

Do you have some better ideas?

thx Michi

**Comment by stephenberry** [comment]
Good question. I've designed the code so that I can easily add customizable context classes in the future. I think this would be the best approach, this way you could add any custom fields to your context.

I'll open an issue to formally add this feature. Right now you could use `current_file`, because you probably aren't using file includers in Glaze.

**Comment by stephenberry** [comment]
`std::string_view custom_error_message` has been added to `glz::error_ctx` and `glz::context` in #1150
